src/compile.rs

```diff
@@ -329,6 +346,128 @@ pub fn build_func_map<'a>(
     Ok(decl_map)
 }
 
+// Helper function to deal with the case where we need to either create a
+// new error or add to an existing one, but specifically for this issue
+// we need to create a new error and immediately add to it.
+#[allow(clippy::too_many_arguments)]
+fn new_error_helper(
+    error: Option<CompileError>,
+    msg: &str,
+    file_a: &SimpleFile<String, String>,
+    file_b: &SimpleFile<String, String>,
+    range_a: Range<usize>,
+    range_b: Range<usize>,
+    help_a: &str,
+    help_b: &str,
+) -> CompileError {
+    let mut tmp_error;
```

Comment:
Now that this is in its own function (which is a huge improvement, IMO.  Thanks), this variable is probably better named "ret".